sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HL7v2IO.java

```diff
@@ -397,7 +492,8 @@ public void instantiateHealthcareClient() throws IOException {
         public void processElement(ProcessContext context) {
           String msgId = context.element();
           try {
-            context.output(HL7v2Message.fromModel(fetchMessage(this.client, msgId)));
+            HL7v2Message msg = HL7v2Message.fromModel(fetchMessage(this.client, msgId));
+            context.output(TimestampedValue.of(msg, Instant.parse(msg.getSendTime())));
```

Comment:
You should really be using `context.outputWithTimestamp` and not `TimestampedValue`.

It would make sense to have an object that chooses which timestamp to use (input elements timestamp or messages send time) instead of having a different variant of the DoFn. In both cases you would use the monotonically increasing estimator.